src/Text/Parser/TreeSitter/Language.hs

```diff
@@ -32,17 +32,24 @@ class Symbol s where
 -- | TemplateHaskell construction of a datatype for the referenced Language.
 mkSymbolDatatype :: Name -> Ptr Language -> Q [Dec]
 mkSymbolDatatype name language = do
-  symbols <- runIO $ languageSymbols language
+  symbols <- ((Regular, "ParseError") :) <$> runIO (languageSymbols language)
+  let namedSymbols = uncurry symbolToName <$> symbols
 
   Module _ modName <- thisModule
   pure
-    [ DataD [] name [] Nothing (flip NormalC [] . mkName . uncurry symbolToName <$> symbols) [ ConT ''Show, ConT ''Eq, ConT ''Enum, ConT ''Ord, ConT ''Bounded, ConT ''Ix ]
-    , InstanceD Nothing [] (AppT (ConT ''Symbol) (ConT name)) [ FunD 'symbolType (uncurry (clause modName) <$> symbols) ] ]
+    [ DataD [] name [] Nothing (flip NormalC [] . mkName <$> namedSymbols) [ ConT ''Show, ConT ''Eq, ConT ''Ord, ConT ''Bounded, ConT ''Ix ]
+    , InstanceD Nothing [] (AppT (ConT ''Symbol) (ConT name)) [ FunD 'symbolType (uncurry (clause modName) <$> symbols) ]
+    , InstanceD Nothing [] (AppT (ConT ''Enum) (ConT name))
+      [ FunD 'toEnum (zipWith (toEnumClause modName) [-1..] namedSymbols)
+      , FunD 'fromEnum (zipWith (fromEnumClause modName) [-1..] namedSymbols) ] ]
```

Comment:
tree-sitter’s magic `ERROR` symbol is `-1`, which is reflected in the `Enum` instance defined here.